libbeat/outputs/logstash/enc.go

```diff
@@ -24,7 +24,11 @@ import (
 
 func makeLogstashEventEncoder(info beat.Info, escapeHTML bool, index string) func(interface{}) ([]byte, error) {
 	enc := json.New(false, escapeHTML, info.Version)
-	return func(event interface{}) ([]byte, error) {
-		return enc.Encode(index, event.(*beat.Event))
+	return func(event interface{}) (d []byte, err error) {
+		d, err = enc.Encode(index, event.(*beat.Event))
+		if err != nil {
+			debugf("Failed to encode event: %v", event)
```

Comment:
We should still include an error log here, as we have for the other outputs. Otherwise it's most likely to get unnoticed.

Reply:
There are already errors logged for all events failures for both implementations ([sync](https://github.com/elastic/beats/blob/9748535fe20a65e7e99597067544f96fa775c6f5/libbeat/outputs/logstash/sync.go#L155) and [async](https://github.com/elastic/beats/blob/9748535fe20a65e7e99597067544f96fa775c6f5/libbeat/outputs/logstash/async.go#L256)). But I wanted to add debug logging only for events failed to encode, and unlike the other outputs, on this one it is not easy to differentiate if it was an encoding event at the point where the general message is logged. This is why I add this debug message here.